frontend/src/v5/store/teamspaces/teamspaces.sagas.ts

```diff
@@ -58,15 +59,39 @@ export function* fetchQuota({ teamspace }) {
 	}
 }
 
-export  function* waitForAddons() {
+export function* waitForAddons() {
 	const isFetchingAddons = yield select(selectIsFetchingAddons);
 
 	if (isFetchingAddons) {
 		yield take(TeamspacesTypes.FETCH_ADD_ONS_SUCCESS);
 	}
 }
 
+export function* fetchActivityLog({ teamspace, startDate, endDate }) {
+	try {
+		const log = yield API.Teamspaces.fetchActivityLog(teamspace, startDate, endDate);
+		const url = window.URL.createObjectURL(new Blob([log]));
+		const link = document.createElement('a');
+		link.href = url;
+		const range = [startDate, endDate].filter(Boolean).map((date) => formatFilenameDate(date)).join('__');
+		link.download = formatMessage({ id: 'teamspaces.fetchActivityLog.filename', defaultMessage: 'activity_log_{range}.zip' }, { range });
+		document.body.appendChild(link);
+		link.click();
+		document.body.removeChild(link);
```

Comment:
Is there a reason you can't use `downloadFile` from `src\v5\ui\components\authenticatedResource\authenticatedResource.hooks.ts` ? (which at this point I think should go to a helper file along with `deleteAuthUrlFromCache` and `downloadAuthUrl` because they  seem more like helpers)